src/query/stream/streams.cpp

```diff
@@ -724,12 +725,16 @@ TransformationResult Streams::Check(const std::string &stream_name, std::optiona
           auto accessor = interpreter_context->db->Access();
           CallCustomTransformation(transformation_name, messages, result, accessor, *memory_resource, stream_name);
 
-          for (auto &row : result.rows) {
+          for (auto idx = 0; idx < result.rows.size(); ++idx) {
+            const auto &row = result.rows[idx];
+            const auto &message = messages[idx];
```

Comment:
This made me think. If I am not mistaken, it is not guaranteed that the number of queries are exactly the same as the number of messages:
 - in case of merging multiple updates into one query, the number of queries might be less than the number of messages
 - in case of "multi step" updates, the number of queries might be greater than the number of messages
 
I propose to yield `max(result.rows.size(), messages.size())` rows from the `CHECK` query properly handling the possible differences in the number of messages/queries.

Reply:
Correct, it is not guaranteed. There can be `N` messages and `M` queries. It all depends on the user and how the transformation is defined.

Reply:
So I have implemented as follow:
1. size(queries) == size(messages): We display 1 message per query (current implementation).
2. We create a list from all messages and show only for the first query.

An example of what it would look like if the transformation would do some filtering (thus create less queries than it has received messages):
![example_transformation_filtering_messages](https://user-images.githubusercontent.com/97525434/167851522-aed520c6-6bc1-4853-b779-307db9028b6b.png)

Reply:
@42jeremy I don't really like this because in the second case the last column of the first row is filled with a bunch of messages and the rest is plain null -- it doesn't reason well to me to have the last column of each row (other than the first) being null. Consider Benjo's approach.

@antaljanosbenjamin even with your approach, if `result.rows.size() < messages.size()` then max would produce as many rows as the size of the messages. Why not limit the number of raw messages to the size of the result queries? 

All in all, I think there is a general problem with the current approach: we don't know (and we can't know) the relationship between the number of messages and the queries. Therefore, why don't we separate these by adding a `CHECK STREAM RAW` statement instead (which basically returns N rows containing N raw messages from the stream)?

Reply:
> Why not limit the number of raw messages to the size of the result queries?

Two reasons:
1. Most probably the query and the message on the same line are not related to each other.
2. It might show a query that was created from a message that is not shown, thus really hard to debug. In this way we at least show everything in the "same" order, so the user can debug it based on the actual data.

> All in all, I think there is a general problem with the current approach: we don't know (and we can't know) the relationship between the number of messages and the queries.

100 percent agree.

> Therefore, why don't we separate these by adding a CHECK STREAM RAW statement instead (which basically returns N rows containing N raw messages from the stream)?

We can do that, if that is better in your opinion, however that would require the user to run two queries always, not talking about taking care of the same parameters for the `CHECK STREAM` and `CHECK STREAM RAW` queries.

With the current approach, if the `message -> query` is 1:1 mapping, the user gets a super nice view. If not, than get a not that good view, but at least it is done by one query (without caring about the parameters).

Maybe you can discuss this tomorrow or on Friday on daily, or on weekly.

Reply:
During discussing this on daily the main talking points were the following:

1. Even when the number of messages and queries are same, the 1-to-1 mapping is not guaranteed. It is possible that one message might not result in any query and another message might result in two queries. Also, considering 1-to-1 mapping, the order of the messages and queries might not be the same. Therefore showing 1 query and 1 message in a single row might not make sense.
2. Instead of having the messages aggregated into a list when the number of messages and queries don't match, we might want to yield `max(result.rows.size(), messages.size())` rows, but we couldn't decide on that as we didn't see any clear advantage of them.
3. Having two queries (`CHECK STREAM` and `CHECK STREAM RAW`) does not guarantee that the queries are generated from the messages the user sees. If the batch "closes" because of batch interval (the longest time until which a message should wait for other messages to include in the same batch), then the `CHECK STREAM` and `CHECK STREAM RAW` queries might experience different batches. The lower the batch interval, the likely it will happen.

Because of these we have the following questions:
1. Does showing 1 message and query in 1 line makes sense when their numbers are equal?
2. Should we **always** show one message per line, or should we aggregate them? See example below.
3. Should we have separate queries for getting the message and getting the queries from the transformation?

### Example for #2 question

Always show 1 message per line:
```
+--------------------------+------------+----------------------+
| query                    | parameters | raw messages         |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'a'})" | NULL       | "a"                  |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'b'})" | NULL       | "b"                  |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'c'})" | NULL       | "don't create query" |
+--------------------------+------------+----------------------+
| NULL                     | NULL       | "c"                  |
+--------------------------+------------+----------------------+
```
Or when the number of message is less than the queries:

```
+--------------------------+------------+----------------------+
| query                    | parameters | raw messages         |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'a'})" | NULL       | "a"                  |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'a'})" | NULL       | "b"                  |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'b'})" | NULL       | NULL                 |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'b'})" | NULL       | NULL                 |
+--------------------------+------------+----------------------+
```

Versus the aggregated versions:
```
+--------------------------+------------+--------------------------------------+
| query                    | parameters | raw messages         |
+--------------------------+------------+--------------------------------------+
| "CREATE (v {prop: 'a'})" | NULL       | ["a", "b", "don't create query", "c] |
+--------------------------+------------+--------------------------------------+
| "CREATE (v {prop: 'b'})" | NULL       | NULL                                 |
+--------------------------+------------+--------------------------------------+
| "CREATE (v {prop: 'c'})" | NULL       | NULL                                 |
+--------------------------+------------+--------------------------------------+
```
and
```
+--------------------------+------------+----------------------+
| query                    | parameters | raw messages         |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'a'})" | NULL       | ["a", "b"]           |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'a'})" | NULL       | NULL                 |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'b'})" | NULL       | NULL                 |
+--------------------------+------------+----------------------+
| "CREATE (v {prop: 'b'})" | NULL       | NULL                 |
+--------------------------+------------+----------------------+
```

Reply:
We have settled to use the format @42jeremy presented in [his comment](https://github.com/memgraph/memgraph/pull/394#issuecomment-1126184318). The main reason is even with the same number of queries and messages, we cannot be sure about the mapping. By providing batches of queries and messages in each row, we trust the user to do the mapping by themself.